urbanroute/urbanroute/geospatial/intersection.py

```diff
@@ -1,10 +1,52 @@
 """Geospatial intersection between graphs and dataframes."""
 
 import logging
-from typing import Optional
+from typing import Any, Optional
 import networkx as nx
 import osmnx as ox
 import geopandas as gpd
+from sqlalchemy import func
+from cleanair.databases import DBReader
+from cleanair.databases.tables import AirQualityResultTable, OSHighway
+from cleanair.decorators import db_query
+from cleanair.mixins import ResultQueryMixin
+from cleanair.types import Source
+
+
+class RoadQuery(ResultQueryMixin, DBReader):
+    """Class for querying air quality results on roads."""
+
+    @property
+    def result_table(self) -> AirQualityResultTable:
+        """The sqlalchemy table to query."""
+        return AirQualityResultTable
+
+    @db_query
+    def pollution_on_roads(self, instance_id: str, timestamp: str) -> Any:
+        """Map air pollution forecasts on the hexgrid to the road network."""
+        # use a subquery to get the forecasts from the result table
+        result_sq = self.query_results(
+            instance_id, Source.hexgrid, output_type="subquery"
+        )
+
+        with self.dbcnxn.open_session() as session:
+            roads = (
+                # take the mean and variance of each hexgrid cell the road passes through
+                session.query(
+                    OSHighway,
+                    (func.avg(result_sq.c.NO2_mean) * OSHighway.length).label(
```

Comment:
Could provide an argument to choose how pollution is calculated. Maybe move to future PR